src/custom_request_schedule_manager.cc

```diff
@@ -52,62 +61,35 @@ CustomRequestScheduleManager::CustomRequestScheduleManager(
           params.batch_size, params.measurement_window_ms, params.max_trials,
           params.max_threads, params.num_of_sequences,
           params.shared_memory_type, params.output_shm_size,
-          params.serial_sequences, parser, factory, params.request_parameters)
+          params.serial_sequences, parser, factory, params.request_parameters),
+      warmup_request_count_(params.warmup_request_count),
+      benchmark_request_count_(params.request_count)
 {
-  max_threads_ = std::min(max_threads_, params.request_count);
-}
-
-cb::Error
-CustomRequestScheduleManager::InitCustomSchedule(const size_t request_count)
-{
-  PauseWorkers();
-  ConfigureThreads(request_count);
-  GenerateSchedule();
-  ResumeWorkers();
-
-  return cb::Error::Success;
 }
 
 void
-CustomRequestScheduleManager::GenerateSchedule()
+CustomRequestScheduleManager::DistributeScheduleToWorkers(
+    const Schedule& schedule)
 {
-  auto worker_schedules = CreateWorkerSchedules(schedule_);
+  auto worker_schedules = CreateWorkerSchedules(schedule);
   GiveSchedulesToWorkers(worker_schedules);
 }
 
-std::vector<RateSchedulePtr_t>
-CustomRequestScheduleManager::CreateWorkerSchedules(
-    const std::vector<std::chrono::milliseconds>& schedule)
-{
-  std::vector<RateSchedulePtr_t> worker_schedules =
-      CreateEmptyWorkerSchedules();
-  std::vector<size_t> thread_ids{CalculateThreadIds()};
-  size_t thread_id_index = 0;
-  size_t worker_index = 0;
-
-  for (const auto& timestamp : schedule) {
-    const auto timestamp_ns{
-        std::chrono::duration_cast<std::chrono::nanoseconds>(timestamp)};
-    worker_index = thread_ids[thread_id_index];
-    thread_id_index = ++thread_id_index % thread_ids.size();
-    worker_schedules[worker_index]->intervals.push_back(timestamp_ns);
-  }
-  SetScheduleDurations(worker_schedules);
-
-  return worker_schedules;
-}
-
 void
 CustomRequestScheduleManager::InitManagerFinalize()
 {
-  schedule_ = GetScheduleFromDataset();
+  auto [warmup_schedule, benchmark_schedule]{GetSchedulesFromDataset()};
+  warmup_schedule_ = std::move(warmup_schedule);
+  benchmark_schedule_ = std::move(benchmark_schedule);
```

Comment:
As mentioned, there's now two phases: warmup and benchmark, so there's two schedules that get stored and used during the benchmark session.